pkg/istio/authorizationpolicy/controller.go

```diff
@@ -0,0 +1,549 @@
+					log.Infoln("Deleting Authz Policy ... ")
+					c.configStoreCache.Delete(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace)
+				}
+			}
+		} else if castItem.Operation == model.EventUpdate {
+			if _, ok = obj.Annotations[authzEnabledAnnotation]; ok {
+				if obj.Annotations[authzEnabledAnnotation] == authzEnabled {
+					if res := c.configStoreCache.Get(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace); res == nil {
+						log.Infof("istio authz annotation for service %s is updated to true\n", obj.Name)
+						// form the authorization policy config
+						athenzDomainRaw, exists, err := c.adIndexInformer.GetIndexer().GetByKey(athenz.NamespaceToDomain(obj.Namespace))
+						if err != nil {
+							return err
+						}
+
+						if !exists {
+							// TODO, add the non existing athenz domain to the istio custom resource
+							// processing controller to delete them
+							return fmt.Errorf("athenz domain %s does not exist in cache", obj.Namespace)
+						}
+
+						athenzDomain, ok := athenzDomainRaw.(*adv1.AthenzDomain)
+						if !ok {
+							return errors.New("athenz domain cast failed")
+						}
+
+						signedDomain := athenzDomain.Spec.SignedDomain
+						labels := obj.GetLabels()
+						domainRBAC := m.ConvertAthenzPoliciesIntoRbacModel(signedDomain.Domain, &c.adIndexInformer)
+						convertedCR = c.convertAthenzModelIntoIstioAuthzPolicy(domainRBAC, obj.Namespace, obj.Name, labels["svc"])
+						log.Infoln("Creating Authz Policy ... ")
+						revision, err := c.configStoreCache.Create(convertedCR)
+						if err != nil {
+							log.Errorln("error creating authz policy: ", err.Error())
+							return err
+						}
+						log.Infoln("Revision number is: ", revision)
+					}
+				} else {
+					// case when service has authz flag switch from true to false, authz policy with the same name present
+					if res := c.configStoreCache.Get(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace); res != nil {
+						log.Infoln("Deleting Authz Policy ... ")
+						c.configStoreCache.Delete(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace)
+					}
+				}
+			} else {
+				// case when service has authzEnabledAnnotation removed, and authz policy with the same name present
+				if res := c.configStoreCache.Get(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace); res != nil {
+					log.Infoln("Deleting Authz Policy ... ")
+					c.configStoreCache.Delete(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace)
+				}
+			}
+		} else if castItem.Operation == model.EventDelete {
+			log.Infoln("Deleting Authz Policy ... ")
+			err = c.configStoreCache.Delete(schemas.AuthorizationPolicy.Type, obj.Name, obj.Namespace)
+			if err != nil {
+				return err
+			}
+		}
+	} else if obj, ok := (castItem.Resource).(*adv1.AthenzDomain); ok {
+		// athenz domain update will result in update in the authz policies in the corresponding namespace
+		if castItem.Operation == model.EventUpdate {
```

Comment:
Aren't we missing the create / delete events here as well?

Reply:
delete events are rarely seen for athenz domain crd. When there is such event, it usually means namespace is deleted, all the resources with the namespace are deleted. We can think of the use case if we want to have delete events added here later.